### PR TITLE
Patterns phase 2

### PR DESCRIPTION
**Pull request by andy-piccalilli**
This is the continuation of #6006 and features:

-  Pagination
-  Make checks the default in status list pattern
-  Button interactive state to match material design spec
-  Stats
-  Details
-  Authors
-  Compare
-  Code blocks
-  Tooltips
-  Form Fields (with material interactions) 
-  Browser support indicators (see codepatterns stuff)

https://deploy-preview-6140--web-dev-staging.netlify.app/design-system/

### CODE REVIEW

File: src/pattern-library/patterns/authors/authors.njk [hunk 1]
```diff
@@ -1,5 +1,5 @@
-<div class="w-author">
-  <a href="{{ data.url }}">
+<div class="author">
+  <a class="avatar" href="{{ data.url }}">
     <img
```
Comment (samthor) [inline src/pattern-library/patterns/authors/authors.njk:3]: I know this isn't part of the review, why not use our shortcode here?

File: src/pattern-library/patterns/authors/authors.njk [hunk 2]
```diff
@@ -17,17 +17,16 @@
...
-    style="display: flex; flex-direction: column; justify-content: center"
-  >
-    <cite class="w-author__name">
+  <div class="flow">
+    <cite class="author__name">
       <a href="{{ data.url }}" class="w-author__name-link">{{ data.name }}</a>
     </cite>
-    <ul class="w-author__link-list">
+    {% if data.bio %}
+      <p class="author__bio">Lectus vivamus aliquet ullamcorper iaculis nisl mauris</p>
```
Comment (samthor) [inline src/pattern-library/patterns/authors/authors.njk:27]: Use the actual bio value?

File: src/pattern-library/patterns/compare/compare.njk [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
+<div class="flow flow-space-size-2">
+  {% for item in data.items %}
+  <figure class="compare flow" data-type="{{ item.type }}">
+    <pre class="compare__code">{{ item.code | md | safe }}</pre>
+    <figcaption class="compare__caption">
+      <p class="compare__label">
+        {{ item.label }}
+      </p>
+      {% if item.caption %}
+        <p>— {{ item.caption | md | safe }}</p>
```
Comment (samthor) [inline src/pattern-library/patterns/compare/compare.njk:10]: why a dash here?

File: src/pattern-library/patterns/tooltips/tooltips.njk [hunk 4]
```diff
@@ -1,5 +1,6 @@
-<button class="w-button--icon" data-icon="{{ data.icon }}">
-  <span class="w-tooltip {{ data.modifier }}" role="tooltip">{{
-    data.tooltip
-  }}</span>
-</button>
+<div class="tooltip" data-alignment="{{ data.modifier }}">
+  <button class="fab" aria-labelledby="{{ data.tooltipID }}">
+    {% include "icons/plus.svg" %}
```
Comment (samthor) [inline src/pattern-library/patterns/tooltips/tooltips.njk:3]: `includeRaw` includes without trying to process the SVG with Nunjucks

File: src/site/_data/design/patterns.js [hunk 5]
```diff
@@ -248,18 +248,29 @@ module.exports = {
...
         },
         rendered: item.rendered,
       });
 
       if (item.variants) {
         item.variants.forEach((variant) => {
-          response.push(variant);
+          response.push({
+            ...variant,
+            ...{
```
Comment (samthor) [inline src/site/_data/design/patterns.js:266]: this `...{` is redundant and confusing to read, move the `data` key up

File: src/site/_data/design/patterns.js [hunk 6]
```diff
@@ -248,18 +248,29 @@ module.exports = {
...
+              data: {
+                title: variant.data.title,
+                extraSampleCSS:
+                  variant.data.extraSampleCSS || item.data.extraSampleCSS,
```
Comment (samthor) [inline src/site/_data/design/patterns.js:270]: We can safely use `??` now if that's more accurate.

File: gulp-tasks/convert-design-tokens.js [hunk 7]
```diff
@@ -53,7 +53,9 @@ const convertDesignTokens = (cb) => {
 
   tokens.colors.forEach((group) => {
     group.items.forEach((color) => {
-      result += `'${nameSlug(group.group + '-' + color.name)}': ${color.hex},`;
+      result += `'${nameSlug(group.group + '-' + color.name)}': ${
+        color.value.includes('data') ? `url("${color.value}")` : color.value
```
Comment (samthor) [inline gulp-tasks/convert-design-tokens.js:57]: Surely these aren't all colors now (it's the section for colors / `token.color`) if a bunch of them are actually SVG backgrounds?

Also, can you change`.includes` => `.startsWith("data:")`) ?

File: src/pattern-library/patterns/authors/authors.njk (hunk 1 above)
Comment (andy-piccalilli) [reply src/pattern-library/patterns/authors/authors.njk:3]: Not that part of the project yet and the current shortcodes still have a lot of hangover from the older stuff (hence the `w-` prefix)

File: src/pattern-library/patterns/authors/authors.njk (hunk 2 above)
Comment (andy-piccalilli) [reply src/pattern-library/patterns/authors/authors.njk:27]: Good catch

File: src/pattern-library/patterns/compare/compare.njk (hunk 3 above)
Comment (andy-piccalilli) [reply src/pattern-library/patterns/compare/compare.njk:10]: Punctuation

File: src/pattern-library/patterns/tooltips/tooltips.njk (hunk 4 above)
Comment (andy-piccalilli) [reply src/pattern-library/patterns/tooltips/tooltips.njk:3]: Yeh I tried to work the icon macros and pulling in the custom blocks was a pain in the ass and taking up too much time. We'll sort in integration.

File: src/scss/mixins/_button-base-styles.scss [hunk 8]
```diff
@@ -88,4 +58,41 @@
...
+      radial-gradient(
+        circle,
+        transparent 1%,
+        get-utility-value('bg', 'action-bg-hover') 1%
+      )
+      center/15000%;
+  }
+
+  &:hover {
+    transform: translateY(-0.15em);
```
Comment (robdodson) [inline src/scss/mixins/_button-base-styles.scss:86]: I was looking at [the material spec](https://material.io/components/buttons) and it doesn't seem like their button does a translate.

**Comment by robdodson** [comment]
A quick note on the browser compat widget. I think there are two designs that Jiwoong did for this, one for articles and one for code patterns.

**Articles**
image

**Code Patterns**
image

The Code Patterns one is smaller and more spaced out. But I think right now the Code Patterns one is the only one implemented. My hunch is that the larger one for articles may look a bit nicer in prose. It shouldn't block this PR but I'm curious if we want to try to support both styles. cc @devnook for their thoughts.

**Review by robdodson — APPROVED** [review]
LGTM but deferring to @samthor for final sign off.

File: src/scss/mixins/_button-base-styles.scss (hunk 8 above)
Comment (andy-piccalilli) [reply src/scss/mixins/_button-base-styles.scss:86]: Ah shoot, my bad. Good catch

**Comment by andy-piccalilli** [comment]
I'm pretty confident the browser compat widget could be done with a variant. Most of the look and feel is applied with utilities, anyway. I'll add it to this run of patterns.

**Comment by andy-piccalilli** [comment]
It would be good to get this landed today or early tomorrow so I can merge it into the `patterns-phase-3` branch 👍

**Pull request merged**